### DISCUSSION TITLE
poem-lambda documentation may need updating

### DISCUSSION THREAD

**Discussion by sccheruku** [poem-web/poem#277] [General]
Bit strapped for time so I explained the issue in the loom video. 

<a href="https://www.loom.com/share/dae35e9f74c547bca041fee9d994f7ae">
    <p>poem/main.rs at master · poem-web/poem - 16 May 2022 - Watch Video</p>
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/dae35e9f74c547bca041fee9d994f7ae-with-play.gif">
  </a>

Basically the examples for [lambda-hello-world](https://github.com/poem-web/poem/blob/master/examples/poem/lambda-hello-world/src/main.rs#L16) shows that we set up routing without the AWS ApiGateway stage name. However, in practice, the stage name needs to be included in the routing for the requests to resolve correctly. 

By the way, I think this is a pretty awesome library.  :) Great work.

**Comment by sccheruku** [comment]
I think the fix for api gateway stage name is pretty simple, we can nest everything inside of a route matching the stage name. example below. 


```rust 

#[tokio::main]
async fn main() -> Result<(), Error> {
    if std::env::var_os("RUST_LOG").is_none() {
        std::env::set_var("RUST_LOG", "poem=debug");
    }
    tracing_subscriber::fmt::init();

    let app = Route::new()
        .at("/", get(simple))
        .at("/pet", get(pet))
        .at("/hello/:name", get(hello));

    let stage_name = String::from("/prod");
    let server = Route::new().nest(stage_name, app);
    run(server).await
}
```

**Comment by sunli829** [comment]
Thank you so much! 🙂
